src/main/scala/firrtl/passes/ReplaceSubAccess.scala

```diff
@@ -0,0 +1,30 @@
+package firrtl.passes
+
+import firrtl.ir._
+import firrtl.{WRef, WSubAccess, WSubIndex, WSubField}
+import firrtl.Mappers._
+import firrtl.Utils._
+import firrtl.WrappedExpression._
+import firrtl.Namespace
+import scala.collection.mutable
+
+
+/** Removes all [[firrtl.WSubAccess]] from circuit
+  */
+object ReplaceAccesses extends Pass {
+  def name = "Replace Accesses"
+
+  def run(c: Circuit): Circuit = {
+    def onStmt(s: Statement): Statement = s map onStmt map onExp
+    def onExp(e: Expression): Expression = e match {
+      case WSubAccess(e, UIntLiteral(value, width), t, g) => WSubIndex(e, value.toInt, t, g)
+      case e => e map onExp
+    }
+  
+    val newModules = c.modules.map( _ match {
+      case m: ExtModule => m
+      case Module(i, n, ps, b) => Module(i, n, ps, onStmt(b))
+    })
+    Circuit(c.info, newModules, c.main)
```

Comment:
no need to match on a map since a match is implicit, could be written as:
```scala
val newModules = c.modules map {
  case m: ExtModule => m
  case Module(i, n, ps, b) => Module(i, n, ps, onStmt(b))
}
```
I'm also kind of on the fence on how I feel about unapply when mapping something to basically the same thing when it has so many fields. The Module match *could* be written as
```scala
case m: Module => m.copy(body = onStmt(m.body))
```
I'm not saying it should be changed to that, just pointing out a possible alternative way to write it.